src/librbd/image/RemoveRequest.h

```diff
@@ -15,11 +15,11 @@ class ProgressContext;
 
 namespace image {
 
-template<typename ImageCtxT = ImageCtx>
+template<typename ImageCtxT = ImageCtx, typename ContextWQT = ContextWQ>
```

Comment:
This shouldn't be required -- you can derive the ContextWQ type if required using traits (see Journal traits for an example).